packages/wrangler/src/__tests__/configuration.test.ts

```diff
@@ -45,6 +46,73 @@ describe("readConfig()", () => {
 			);
 		}
 	});
+
+	describe("environment warnings and errors", () => {
+		const std = mockConsoleMethods();
+
+		it("should warn if an environment is requested but none are defined", () => {
+			writeWranglerConfig();
+			readConfig({ config: "wrangler.toml", env: "foo" });
+			expect(std.warn).toMatchInlineSnapshot(`
+				"[33m▲ [43;33m[[43;30mWARNING[43;33m][0m [1mProcessing wrangler.toml configuration:[0m
+
+				    - No environment found in configuration with name \\"foo\\".
+				      Before using \`--env=foo\` there should be an equivalent environment section in the
+				  configuration.
+
+				      Consider adding an environment configuration section to the wrangler.toml file:
+				      \`\`\`
+				      [env.foo]
+				      \`\`\`
+
+
+				"
+			`);
+		});
+
+		it("should not warn if an environment is requested but none are defined, and `hideWarningForEnvironmentWhenOnlyTopLevel` is true", () => {
+			writeWranglerConfig();
+			readConfig(
+				{ config: "wrangler.toml", env: "foo" },
+				{ hideWarningForEnvironmentWhenOnlyTopLevel: true }
+			);
+			expect(std.warn).toMatchInlineSnapshot(`""`);
+		});
+
+		it("should error if an environment is requested but at least one other is defined", () => {
+			writeWranglerConfig({
+				env: {
+					bar: {},
+				},
+			});
+			expect(() => readConfig({ config: "wrangler.toml", env: "foo" }))
+				.toThrowErrorMatchingInlineSnapshot(`
+				[Error: Processing wrangler.toml configuration:
+				  - No environment found in configuration with name "foo".
+				    Before using \`--env=foo\` there should be an equivalent environment section in the configuration.
+				    The available configured environment names are: ["bar"]
+
+				    Consider adding an environment configuration section to the wrangler.toml file:
+				    \`\`\`
+				    [env.foo]
+				    \`\`\`
+				]
+			`);
+		});
+
+		it("should not error if an environment in `optionalEnvironments` is requested but at least one other is defined, and not update the worker name", () => {
+			writeWranglerConfig({
+				env: {
+					bar: {},
+				},
+			});
+			const config = readConfig(
+				{ config: "wrangler.toml", env: "foo" },
+				{ optionalEnvironments: ["foo"] }
+			);
+			expect(config.name).toEqual("test-name");
+		});
+	});
```

Comment:
It might be clearer to split this into two separate tests, one to check that no warnings are logged and one to check the worker name is not updated.

Reply:
Done